src/libANGLE/renderer/d3d/d3d11/winrt/SwapChainNativeWindow.cpp

```diff
@@ -0,0 +1,182 @@
+//
+// Copyright 2014 The ANGLE Project Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+//
+
+// SwapChainNativeWindow.cpp: NativeWindow for managing IDXGISwapChain native window types.
+
+#include "libANGLE/renderer/d3d/d3d11/winrt/SwapChainNativeWindow.h"
+
+#include <math.h>
+#include <algorithm>
+
+using namespace ABI::Windows::Foundation::Collections;
+using namespace Microsoft::WRL;
+
+namespace
+{
+DXGI_FORMAT GetSwapChainFormat(const ComPtr<IDXGISwapChain> &swapChain)
+{
+    DXGI_SWAP_CHAIN_DESC desc = {};
+    HRESULT result            = swapChain->GetDesc(&desc);
+    if (SUCCEEDED(result))
+    {
+        return desc.BufferDesc.Format;
+    }
+    return DXGI_FORMAT_UNKNOWN;
+}
+}  // namespace
+
+namespace rx
+{
+bool SwapChainNativeWindow::initialize(EGLNativeWindowType window, IPropertySet *propertySet)
+{
+    mSupportsSwapChainResize = false;
+
+    ComPtr<IPropertySet> props = propertySet;
+    ComPtr<IInspectable> win   = window;
+    SIZE swapChainSize         = {};
+    HRESULT result             = S_OK;
+
+    // IPropertySet is an optional parameter and can be null.
+    // If one is specified, cache as an IMap and read the properties
+    // used for initial host initialization.
+    if (propertySet)
```

Comment:
Do we actually support/handle any of these, i.e. specifying either swapchain size or scale?

Reply:
I decided to keep the following variables at false to keep things simple (e550c85dd7340c0e59bbd35c375ef26cc8f983a0):
 - `mSupportsSwapChainResize`
 - `mSwapChainSizeSpecified`
 - `mSwapChainScaleSpecified`